packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -63,38 +97,43 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
     /// @param data       Encoded update data.
     event ConfigUpdate(uint256 indexed version, UpdateType indexed updateType, bytes data);
 
+    /// @notice The block at which the op-node can start searching for logs from.
+    uint256 public startBlock;
+
     /// @notice Semantic version.
-    /// @custom:semver 1.11.0
-    string public constant version = "1.11.0";
+    /// @custom:semver 1.12.0
+    string public constant version = "1.12.0";
 
     /// @notice Constructs the SystemConfig contract. Cannot set
     ///         the owner to `address(0)` due to the Ownable contract's
     ///         implementation, so set it to `address(0xdEaD)`
-    /// @param _owner             Initial owner of the contract.
-    /// @param _overhead          Initial overhead value.
-    /// @param _scalar            Initial scalar value.
-    /// @param _batcherHash       Initial batcher hash.
-    /// @param _gasLimit          Initial gas limit.
-    /// @param _unsafeBlockSigner Initial unsafe block signer address.
-    /// @param _config            Initial resource config.
-    constructor(
-        address _owner,
-        uint256 _overhead,
-        uint256 _scalar,
-        bytes32 _batcherHash,
-        uint64 _gasLimit,
-        address _unsafeBlockSigner,
-        ResourceMetering.ResourceConfig memory _config
-    ) {
+    constructor() {
         initialize({
-            _owner: _owner,
-            _overhead: _overhead,
-            _scalar: _scalar,
-            _batcherHash: _batcherHash,
-            _gasLimit: _gasLimit,
-            _unsafeBlockSigner: _unsafeBlockSigner,
-            _config: _config
+            _owner: address(0xdEaD),
+            _overhead: 0,
+            _scalar: 0,
+            _batcherHash: bytes32(0),
+            _gasLimit: 1,
+            _unsafeBlockSigner: address(0),
+            _config: ResourceMetering.ResourceConfig({
+                maxResourceLimit: 1,
+                elasticityMultiplier: 1,
+                baseFeeMaxChangeDenominator: 2,
+                minimumBaseFee: 0,
+                systemTxMaxGas: 0,
+                maximumBaseFee: 0
+            }),
+            _batchInbox: address(0),
+            _addresses: SystemConfig.Addresses({
+                l1CrossDomainMessenger: address(0),
+                l1ERC721Bridge: address(0),
+                l1StandardBridge: address(0),
+                l2OutputOracle: address(0),
+                optimismPortal: address(0),
+                optimismMintableERC20Factory: address(0)
+            })
         });
+        Storage.setUint(bytes32(uint256(106)), type(uint256).max);
```

Comment:
What is this line for?

Reply:
Its trying to set the `startBlock` - you are right that we want to set it to `type(uint256).max` but hardcoding an offset isn't ideal. There is the `slot` keyword that can be used which is less brittle

Reply:
okay, but why wouldn't that contradict the comment here? https://github.com/ethereum-optimism/optimism/pull/8772#discussion_r1440095078

Reply:
I think that this approach is a good idea. The alternative would be to have an argument to `_setStartBlock`. Can you use the `slot` keyword (only available in yul) and also add a fat comment describing why the code exists

Reply:
yes! good idea

Reply:
could you clarify why use yul instead of `Storage.setUint(..)` as is used in the rest of the contract?

Reply:
we can still use `Storage.setUint` but we cannot use hardcoded magic numbers, so we need to use the `slot` keyword to replace the magic number

Reply:
got it. thanks!

Reply:
plz check https://github.com/ethereum-optimism/optimism/pull/8772/commits/28b7d7bc280c053bdea84af2054331bd5226466d